core/consensus/aura/src/lib.rs

```diff
@@ -284,10 +294,10 @@ impl<B: Block, C, E, I, Error, SO> SlotWorker<B> for AuraWorker<C, E, I, SO> whe
 			);
 			return Box::new(future::ok(()));
 		}
-
-		let proposal_work = match slot_author(slot_num, &authorities) {
+		let maybe_author: Option<AuthorityId<P>> = slot_author::<P>(slot_num, &authorities);
```

Comment:
Do you really need to specify the type here?
And slot_author should not require the explicit generic parameter?

Reply:
`rustc` cannot infer the type of `maybe_author` unless I provide the type parameter to `slot_author`.